### PR TITLE
Fix  abort bug issue caused by rst stream interrupt the writing

### PR DESCRIPTION
**Pull request by zhimingxie**
The detail of the problem I've written in the issue https://github.com/grpc/grpc/issues/10516

### CODE REVIEW

File: src/core/ext/transport/chttp2/transport/chttp2_transport.c [hunk 1]
```diff
@@ -712,6 +712,7 @@ static void destroy_stream_locked(grpc_exec_ctx *exec_ctx, void *sp,
 
   grpc_chttp2_list_remove_stalled_by_transport(t, s);
   grpc_chttp2_list_remove_stalled_by_stream(t, s);
+  grpc_chttp2_list_remove_writing_stream(t, s);
```
Comment (ctiller) [inline src/core/ext/transport/chttp2/transport/chttp2_transport.c]: This is quite dangerous: a different thread could be writing at this point, and so this is likely to race and crash.

Comment (zhimingxie) [inline src/core/ext/transport/chttp2/transport/chttp2_transport.c]: in my humble opinion, the writing op and destroy_stream_locked both are scheduled by t->combiner  so there won't be race you mentioned, will there ?

Also where do you think it is better to do this remove ?

**Comment by grpc-testing** [comment]
Thanks for your pull request. The automated tests will run as soon as one of the admins verifies this change is ok for us to run on our infrastructure.

**Comment by grpc-testing** [comment]
Thanks for your pull request. The automated tests will run as soon as one of the admins verifies this change is ok for us to run on our infrastructure.

**Comment by thelinuxfoundation** [comment]
Thank you for your pull request. Before we can look at your contribution, we need to ensure all contributors are covered by a Contributor License Agreement.

After the following items are addressed, please respond with a new comment here, and the automated system will re-verify.

* User @zhimingxie isn't covered by a CLA. They will need to complete the form at https://identity.linuxfoundation.org/projects/cncf

_Regards,
CLA GitHub bot_

**Comment by jtattermusch** [comment]
Closing as #10516 has also been closed.

**Pull request closed without merging**